src/pages/Attendance.jsx

```diff
@@ -1,55 +1,241 @@
+      ...prevState,
+      value: newData,
+      isOpenCalendar: false,
+    }))
   }
 
   const handlePrevDayClick = () => {
-    const newDate = dayjs(value).subtract(1, "day")
-    setValue(newDate)
+    setState(prevState => ({
+      ...prevState,
+      value: dayjs(prevState.value).subtract(7, "day"),
+      selectedLeader: null,
+      adultCount: 0,
+      memberCount: 0,
+      totalCount: 0,
+      isChecked: {},
+      attendanceAllData: {},
+    }))
   }
 
   const handleNextDayClick = () => {
-    const newDate = dayjs(value).add(1, "day")
-    setValue(newDate)
+    setState(prevState => ({
+      ...prevState,
+      value: dayjs(prevState.value).add(7, "day"),
+      selectedLeader: null,
+      adultCount: 0,
+      memberCount: 0,
+      totalCount: 0,
+      isChecked: {},
+      attendanceAllData: {},
+    }))
   }
 
   const handleLeaderClick = cell => {
-    setSelectedLeader(cell)
+    setState(prevState => ({
+      ...prevState,
+      selectedLeader: cell,
+    }))
   }
 
-  const handleCheck = e => {
-    setIsChecked(e.target.checked)
-  }
+  const handleCheck = member => {
+    setState(prevState => {
+      const isChecked = { ...prevState.isChecked }
+      isChecked[member] = !isChecked[member]
+
+      let memberCount = prevState.memberCount
+      if (isChecked[member]) memberCount++
+      else memberCount--
+
+      const totalCount = isChecked[member] ? prevState.totalCount + 1 : prevState.totalCount - 1
 
+      updateAttendanceData(state.value.format("YYYY-MM-DD"), state.selectedLeader, member)
+
+      return {
+        ...prevState,
+        isChecked,
+        memberCount,
+        totalCount,
+      }
+    })
+  }
   const handleCalendarClick = () => {
-    setIsOpenCalendar(true)
+    setState(prevState => ({
+      ...prevState,
+      isOpenCalendar: true,
+      selectedLeader: null,
+      adultCount: 0,
+      memberCount: 0,
+      totalCount: 0,
+      isChecked: {},
+      attendanceAllData: {},
+    }))
   }
+
+  const handleAdultData = id => {
+    setState(
+      prevState => {
+        const adultCount = id === "add" ? prevState.adultCount + 1 : prevState.adultCount > 0 ? prevState.adultCount - 1 : 0
+
+        return {
+          ...prevState,
+          adultCount,
+          totalCount: prevState.memberCount + adultCount,
+        }
+      },
+      updateAttendanceData(state.value.format("YYYY-MM-DD"), state.selectedLeader, "adults", id),
+    )
+  }
+
+  const updateAttendanceData = (date, leader, member, id) => {
+    const updatedData = { ...state.attendanceAllData }
+
+    if (!updatedData[date]) {
+      updatedData[date] = {
+        adultAttendance: 0,
+        memberAttendance: 0,
+        cellData: {},
+      }
+    }
+
+    const isLeaderEmpty = !leader
+
+    if (isLeaderEmpty) {
+      if (!updatedData[date].cellData.adults) {
+        updatedData[date].cellData.adults = []
+      }
+      if (member === "adults") {
+        if (id === "add") {
```

Comment:
```suggestion
        if (id === "add") {
```
이건 action만 id로 바꾼거고 방식은 같은거 같은데..? id가 add, remove인게 이상하지 않아?
함수하나로 다 처리하려고하지말고 나누는 방식으로 고민해봐
일반적으로는 함수가 이렇게 길어지는건 안좋아. 지금 보면 이 함수는 매개변수도 4개나되고, 하나의 함수가 하는 역할이 너무 많아. 여러 함수를 만들어도되니까 함수마다 쉽게 이해할 수 있도록 나눠서 함수를 짜봐